projects/rero/ng-core/src/lib/record/search/record-search.service.ts

```diff
@@ -105,77 +122,76 @@ export class RecordSearchService {
   }
 
   /**
-   * Get filters of an aggregation
-   * @param key Aggregation key
+   * Remove a filter from the current aggregations filters.
+   * @param key filter name
+   * @param value filter value
    */
-  getAggregationFilters(key: string): Observable<Array<string>> {
-    return this._aggregationsFiltersSubject.pipe(
-      first(),
-      map((aggregationsFilters: Array<AggregationsFilter>) => {
-        const index = aggregationsFilters.findIndex(item => item.key === key);
-        return index === -1 ? [] : aggregationsFilters[index].values;
-      })
-    );
+  removeFilter(key, value) {
+    const filter = this._aggregationsFilters.find(item => item.key === key);
+    if (filter) {
+      filter.values = filter.values.filter(item => item !== value);
+      // No more selected values, remove key from aggregations filters.
+      if (filter.values.length === 0) {
+        this._aggregationsFilters = this._aggregationsFilters.filter(item => item.key !== filter.key);
+      }
+    }
   }
 
   /**
-   * Removes the given value from selected filters and removes all children
-   * selected values, too.
-   * @param key Aggregation key
-   * @param bucket Bucket containing the value to remove
+   * Check if a given filter exist in the current aggregations filters.
+   * @param key filter name
+   * @param value filter value
+   * @returns true if exists
    */
-  removeAggregationFilter(key: string, bucket: any) {
-    let aggregationsFilters = cloneDeep(this._aggregationsFilters);
-
-    const index = aggregationsFilters.findIndex((item: any) => {
-      return item.key === key;
-    });
-
-    let values = aggregationsFilters[index].values;
-
-    // Remove selected value
-    values = values.filter((selectedValue: string) => {
-      return selectedValue !== bucket.key;
-    });
+   hasFilter(key, value): boolean {
+    const filter = this._aggregationsFilters.find((a) => a.key === key);
+    return !!(filter && filter.values.some((v) => v === value));
```

Comment:
I learned something about the double exclamation mark operator.
Maybe it will be interesting to add a comment to explain this operator ? or mention this unusual operator during a daily.

Reply:
You can find more details here: https://www.samanthaming.com/tidbits/19-2-ways-to-convert-to-boolean/. I would mention this in the daily. I found this syntax a bit ugly but seems to be the best approach.